src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java

```diff
@@ -210,11 +210,15 @@ private String proxyPost(Request request, Response response) {
             } catch(NullPointerException e) {
                 LOG.warn("Failed to read variables from GraphQL Plan request. Still passing to OTP2: {}", e.getMessage());
             }
-
         }
 
-        // provide response to requester as received from OTP server
-        Arrays.stream(otpDispatcherResponse.headers).forEach(header -> response.header(header.getName(), header.getValue()));
+        // Add response headers to requester as it was received from OTP server, but beware that
+        // spark.Response#header() will duplicate rather than update, so filter out duplicates.
+        Arrays.stream(otpDispatcherResponse.headers)
+            .filter(h -> !response.raw().containsHeader(h.getName()))
+            //.peek(h -> LOG.info("NEW {}={}", h.getName(), h.getValue()))
```

Comment:
Remove this line before merging.

Reply:
Done.

Reply:
> Remove this line before merging.

I did and suddenly there are CI failures in the end-2-end testing. This shouldn't triggers those. Rerunning.